codingStandards/CodingStandard-Java.md

```diff
@@ -385,23 +385,26 @@ The package statement location is enforced by the Java language. A Java package
 
 The rule of thumb is to package the classes that are related. For example in Java, the classes related to file writing is grouped in the package `java.io` and the classes which handle lists, maps etc are grouped in `java.util` package.
 
-**2. The import statements must follow the package statement. import statements should be sorted with the most fundamental packages first, and grouped with associated packages together and one blank line between groups.**
+**2. Import statements must follow the package statement.**
 
-```java
-import java.io.IOException;
-import java.net.URL;
-import java.rmi.RmiServer;
-import java.rmi.server.Server;
-import javax.swing.JPanel;
-import javax.swing.event.ActionEvent;
-import org.linux.apache.server.SoapServer;
-```
+The location of import statements is enforced by the Java language.
```

Comment:
Yes, Java enforces this. 

I did some trials, and the result shows that javac will report error if import statements are before the package statement, e.g.

```java
import java.io.PrintWriter;

package test; // Error message: "test/Test.java:3: error: class, interface, or enum expected"

public class Test {
    public static void main(String[] args) {
        PrintWriter pw = new PrintWriter(System.out);
        pw.println("hello world!!");
        pw.flush();
    }  
}
```

javac will also report error if import statement are not right after the package statement, e.g. separated by a class definition.

```java
package test;

class OtherClass {
      // after removing this class the file will successfully compile
}

import java.io.PrintWriter; // Error message: "test/Test.java:7: error: class, interface, or enum expected"

public class Test {
    public static void main(String[] args) {
        PrintWriter pw = new PrintWriter(System.out);
        pw.println("hello world!!");
        pw.flush();
    }  
}
```

So I think yes, this rule can be removed since Java compiler (not Java actually, will submit a fix) enforces this. Should we?